packages/core-macro/src/component_body/utils.rs

```diff
@@ -0,0 +1,56 @@
+use crate::component_body::ComponentBody;
+use dioxus_core::{Element, Scope};
+use quote::ToTokens;
+use syn::{parse_quote, Path};
+
+/// The output produced by a deserializer.
+///
+/// # For implementors
+/// Struct field guidelines:
+/// * Must be public, so that other deserializers can utilize them.
+/// * Should usually be [`Item`]s that you then simply combine in a [`quote!`]
+/// in the [`ComponentBodyDeserializer::output_to_token_stream2`] function.
+/// * If an [`Item`] might not be included, wrap it in an [`Option`].
+/// * Must be at the component function "level"/"context".
+/// For example, the [`InlinePropsDeserializer`](crate::component_body_deserializers::inline_props::InlinePropsDeserializer)
+/// produces two [`Item`]s; the function but with arguments turned into props, and the props struct.
+/// It does not return any [`Item`]s inside the struct or function.
+pub trait DeserializerOutput: ToTokens {}
+
+/// The args passed to a [`ComponentBody`] when deserializing it.
+///
+/// It's also the struct that does the deserializing.
+/// It's called "DeserializerArgs", not "Deserializer". Why?
+/// Because "args" makes more sense to the caller of [`ComponentBody::deserialize`], which
+/// takes an [`DeserializerArgs`] argument. However, you can think of "DeserializerArgs" as the deserializer.
+pub trait DeserializerArgs<TOutput>: Clone
```

Comment:
Deserialize might be a bit consusing for tokens because of how common it is with `serde`. Syn tends to use Parse for tokens maybe we could use ParseArgs?

Reply:
I think deserialize makes more sense. Maybe not in this case, but `ComponentBody::deserialize()` is IMHO better than `ComponentBody::parse()`. It also doesn't parse/deserialize tokens, but rather the `ComponentBody`.